addons/mail/models/mail_link_preview.py

```diff
@@ -42,17 +42,14 @@ def _create_link_previews(self, message):
             link_previews |= self.env['mail.link.preview']._create_link_preview(url, message.id, requests_session)
         if not link_previews:
             return
-        guest = self.env['mail.guest']._get_guest_from_context()
-        if message.model == 'discuss.channel' and message.res_id:
-            target = self.env['discuss.channel'].browse(message.res_id)
-        elif self.env.user._is_public() and guest:
-            target = guest
-        else:
-            target = self.env.user.partner_id
-        self.env['bus.bus']._sendone(target, 'mail.record/insert', {
+        self.env['bus.bus']._sendone(link_previews._bus_notification_target(message), 'mail.record/insert', {
             'LinkPreview': link_previews._link_preview_format()
         })
 
+    @api.model
+    def _bus_notification_target(self, message):
+        return self.env.user.partner_id
```

Comment:
not happy with this being linked to model rather than individual records (like in attachment)

Reply:
not happy with param being used only in override

Reply:
We are thinking of using link preview in knowledge... shouldn't we think of making it a standalone module with res_model / res_id maybe ? or we duplicate the code ?